### PR TITLE
💎   Refactor inbound NAT rules service

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: Uses the new Service interface to break out inbound NAT rule creation / deletion from the network interface service.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Partially addresses #757 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [x] adds unit tests

**Release note**:

```release-note
Refactor inbound NAT rules service
```

### CODE REVIEW

File: cloud/services/groups/service.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ type Service struct {
 	Client
 }
 
-// GroupScope defines the scope interface for a disk service.
+// GroupScope defines the scope interface for a group service.
```
Comment (devigned) [inline cloud/services/groups/service.go:30]: Nice drive by documentation upgrade!

File: cloud/services/inboundnatrules/inboundnatrules_test.go [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+				m.Delete(context.TODO(), "my-rg", "my-public-lb", "azure-md-2").
+					Return(autorest.NewErrorWithResponse("", "", &http.Response{StatusCode: 500}, "Internal Server Error"))
+			},
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+			t.Parallel()
```
Comment (CecileRobertMichon) [inline cloud/services/inboundnatrules/inboundnatrules_test.go:301]: /hold

need to apply #806 here

Comment (CecileRobertMichon) [reply cloud/services/inboundnatrules/inboundnatrules_test.go:303]: done

/hold cancel

**Comment by CecileRobertMichon** [comment]
/assign @devigned @nader-ziada this is ready for another review

fixed the parallelism in tests and re-ran make generate

**Pull request merged**